src/js/models/search/SearchAwardsOperation.js

```diff
@@ -0,0 +1,241 @@
+/**
+ * SearchAwardsOperation.js
+ * Created by michaelbray on 8/7/17.
+ */
+
+import { concat } from 'lodash';
+
+import { rootKeys, timePeriodKeys, agencyKeys, awardAmountKeys }
+    from 'dataMapping/search/awardsOperationKeys';
+
+class SearchAwardsOperation {
+    constructor() {
+        this.keyword = '';
+
+        this.timePeriodType = 'fy';
+        this.timePeriodFY = [];
+        this.timePeriodRange = [];
+
+        this.awardType = [];
+
+        this.awardingAgencies = [];
+        this.fundingAgencies = [];
+
+        this.selectedRecipients = [];
+        this.recipientDomesticForeign = 'all';
+        this.selectedRecipientLocations = [];
+        this.recipientType = [];
+
+        this.selectedLocations = [];
+        this.locationDomesticForeign = 'all';
+
+        this.awardAmounts = [];
+
+        this.selectedAwardIDs = [];
+
+        this.selectedCFDA = [];
+        this.selectedNAICS = [];
+        this.selectedPSC = [];
+        this.pricingType = [];
+        this.setAside = [];
+        this.extentCompeted = [];
+    }
+
+    fromState(state) {
+        this.keyword = state.keyword;
+
+        this.timePeriodFY = state.timePeriodFY.toArray();
+        this.timePeriodRange = [];
+        this.timePeriodType = state.timePeriodType;
+        if (state.timePeriodType === 'dr' && state.timePeriodStart && state.timePeriodEnd) {
+            this.timePeriodRange = [state.timePeriodStart, state.timePeriodEnd];
+            this.timePeriodFY = [];
+        }
+
+        this.awardType = state.awardType.toArray();
+
+        this.awardingAgencies = state.selectedAwardingAgencies.toArray();
+        this.fundingAgencies = state.selectedFundingAgencies.toArray();
+
+        this.selectedRecipients = state.selectedRecipients.toArray();
+        this.recipientDomesticForeign = state.recipientDomesticForeign;
+        this.selectedRecipientLocations = state.selectedRecipientLocations.toArray();
+        this.recipientType = state.recipientType.toArray();
+
+        this.selectedLocations = state.selectedLocations.toArray();
+        this.locationDomesticForeign = state.locationDomesticForeign;
+
+        this.awardAmounts = state.awardAmounts.toArray();
```

Comment:
This does not appear to account for custom ranges, which is a nested object in `awardAmounts`